### ISSUE TITLE
missing structure fields False Positive

### ISSUE THREAD

**Issue by Ratipoka** [rust-analyzer/rust-analyzer#11446]
My main.rs file (just the functions used) which works fine with "cargo run" and "cargo check", it does not output errors, just warnings : 
```
use std::num::NonZeroU32;
use std::time::Instant;
use wgpu::{BufferUsages, MapMode, BufferBindingType};

fn create_buffer(device: &wgpu::Device, size: usize, usages: Vec<wgpu::BufferUsages>) -> wgpu::Buffer {
    let output_buffer_size = size as wgpu::BufferAddress;
    let output_buffer_desc = wgpu::BufferDescriptor {
        label: None,
        size: output_buffer_size,
        usage: if usages.len() == 1 {usages[0]} else {
            let mut out = usages[0];
            for usage in usages.iter().skip(1) {
                out |= *usage;
            }
            out
        },
        mapped_at_creation: true,
    };
    device.create_buffer(&output_buffer_desc)
}

async fn my_run() {
    let mut start_ = Instant::now();
    let mut lap_timer = |e: &str| {
        println!("{} took : {:?}", e, start_.elapsed());
        start_ = Instant::now();
    };

    let instance = wgpu::Instance::new(wgpu::Backends::all());
    let adapter = instance
        .request_adapter(&wgpu::RequestAdapterOptions {
            power_preference: wgpu::PowerPreference::default(),
            compatible_surface: None,
            force_fallback_adapter: false,
        })
        .await
        .unwrap();
    let (device, queue) = adapter
        .request_device(&Default::default(), None)
        .await
        .unwrap();
    
    lap_timer("requesting adapter");

    let size = 40_000;
    let data = vec![0u8; size];
    let buf_data = create_buffer(
        &device,
        size,
        vec![BufferUsages::MAP_WRITE, BufferUsages::COPY_SRC] );
    let mut buf_view = buf_data.slice(..).get_mapped_range_mut();
    
    lap_timer("creating buffer");

    for (i, e) in buf_view.iter_mut().enumerate() {
        *e = data[i];
    }

    lap_timer("setting value to buffer");

    let bind_group_layout = device.create_bind_group_layout(
        &wgpu::BindGroupLayoutDescriptor{
                label: None,
                entries: &[
                    wgpu::BindGroupLayoutEntry{
                        binding: 0,
                        visibility: wgpu::ShaderStages::COMPUTE,
                        ty: wgpu::BindingType::Buffer{
                            ty: BufferBindingType::Storage{read_only: false},
                            has_dynamic_offset: false,
                            min_binding_size: None,
                        },
                        count: None,
                    }
                ]
        });

    let bind_group = device.create_bind_group(
        &wgpu::BindGroupDescriptor{                           // error starts here
            label: None,
            layout: &bind_group_layout,
            entries: &[
                wgpu::BindGroupEntry{
                    binding: 0,
                    resource: wgpu::BindingResource::Buffer(buf_data.as_entire_buffer_binding())
                }
            ],                                                             // error ends here
        }
    );
}

fn main() {
    let now = Instant::now();
    // pollster::block_on(run());
    pollster::block_on(my_run());
    println!("{:?}", now.elapsed());
}
```

my cargo.toml : 
```
[package]
name = "windowless"
version = "0.1.0"
authors = ["Ben Hansen <bhbenjaminhansen@gmail.com>"]
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
image = "0.23"
raw-window-handle = "0.3"
winit = "0.26"
shaderc = "0.7"
cgmath = "0.18"
wgpu = "0.12"
pollster = "0.2"
kludgine = "0.1.0"

[[bin]]
name = "windowless"
path = "src/main.rs"
```

Rust analyzer gives me : 

```
missing structure fields:
- binding
- resource
rust-analyzer(missing-fields)
```

I suspect this is because of the lib.rs file (the file where the struct is in : it has two structs with the same name : 
(in wgpu-0.12.0/src/lib.rs)
```
/// Bindable resource and the slot to bind it to.
#[derive(Clone, Debug)]
pub struct BindGroupDescriptor<'a> {
    /// Slot for which binding provides resource. Corresponds to an entry of the same
    /// binding index in the [`BindGroupLayoutDescriptor`].
    pub binding: u32,
    /// Resource to attach to the binding
    pub resource: BindingResource<'a>,
}

/// Describes a group of bindings and the resources to be bound.
#[derive(Clone, Debug)]
pub struct BindGroupDescriptor<'a> {
    /// Debug label of the bind group. This will show up in graphics debuggers for easy identification.
    pub label: Label<'a>,
    /// The [`BindGroupLayout`] that corresponds to this bind group.
    pub layout: &'a BindGroupLayout,
    /// The resources to bind to this bind group.
    pub entries: &'a [BindGroupEntry<'a>],
}
```
PS I have cloned the repo at : https://github.com/sotrh/learn-wgpu and am working in the learn-wgpu-master\docs\showcase\windowless folder, I've just replaced the main.rs file with the main.rs file I showed higher

**Comment by bjorn3** [comment]
The first one should be called `BindGroupEntry`: https://docs.rs/wgpu/0.12.0/src/wgpu/lib.rs.html#1231

Did you accidentally rename it? Cargo doesn't automatically rebuild when the source of a crate from crates.io is changed as it expects it to be immutable. Try removing the `~/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.12.0` directory to let cargo download the source again.

**Comment by Ratipoka** [comment]
Thank you so much ! That was the problem. I was wondering why there was a non existent struct.
